kyaml/yaml/rnode.go

```diff
@@ -382,6 +1124,12 @@ func (rn *RNode) SetYNode(node *yaml.Node) {
 	*rn.value = *node
 }
 
+// Kind returns the kind, if it exists, else empty string.
+func (o *KubeObject) Kind() string {
+	return o.rNode.GetKind()
+}
+
+// Deprecated: Please use rn.AsKubeObject().Kind().
```

Comment:
From user's perspective, this deprecation seems unintuitive. "why should I switch from a simple one function call to a more complicated two-func calls and understand the new term KubeObject"?  Suggest adding a link/brief explanation e.g. distinguish typed object (KubeObj) from untyped object (RNode)

Reply:
Updated